spec/ascii-art-spec.coffee

```diff
@@ -7,15 +6,29 @@ AsciiArt = require '../lib/ascii-art'
 # or `fdescribe`). Remove the `f` to unfocus the block.
 
 describe "AsciiArt", ->
-  promise = null
+  [workspaceElement, activationPromise] = []
+
   beforeEach ->
-    atom.workspaceView = new WorkspaceView()
-    atom.workspace = atom.workspaceView.model
-    promise = atom.packages.activatePackage('ascii-art')
+    workspaceElement = atom.views.getView(atom.workspace)
+    activationPromise = atom.packages.activatePackage('ascii-art')
+
     waitsForPromise ->
       atom.workspace.open()
 
   it "converts", ->
-    atom.workspaceView.trigger 'ascii-art:convert'
+    editor = atom.workspace.getActiveTextEditor()
+    editor.insertText("yo")
+    editor.selectAll()
+    changeHandler = jasmine.createSpy('changeHandler')
+    editor.onDidChange(changeHandler)
+
+    atom.commands.dispatch workspaceElement, 'ascii-art:convert'
+
     waitsForPromise ->
-      promise
+      activationPromise
+
+    waitsFor ->
+      changeHandler.callCount > 0
+
+    runs ->
+      expect(editor.getText()).toEqual "\n                 \n                 \n __  __    ___   \n/\\ \\/\\ \\  / __`\\ \n\\ \\ \\_\\ \\/\\ \\L\\ \\\n \\/`____ \\ \\____/\n  `/___/> \\/___/ \n     /\\___/      \n     \\/__/       \n"
```

Comment:
@nathansobo Yeah, I wasn't sure how to make it clear, but you're definitely right. 

The expected string is this:

```
       __  __    ___   
      /\ \/\ \  / __`\ 
      \ \ \_\ \/\ \L\ \
       \/`____ \ \____/
        `/___/> \/___/ 
           /\___/      
           \/__/       
```

But that has `\` in it, so I need to escape them, and get this:

```
expected = """
                 
                 
 __  __    ___   
/\\ \\/\\ \\  / __`\\ 
\\ \\ \\_\\ \\/\\ \\L\\ \\
 \\/`____ \\ \\____/
  `/___/> \\/___/ 
     /\\___/      
     \\/__/       
"""
```

That was already a bit unclear, so I put it all in one line. I'd be fine with using that approach if you agree, but would be :joy: if there's a way to write that string without escaping (I guess what I'm asking for is a literal string, not sure if CoffeeScript has something it its toolbox to help with that?).